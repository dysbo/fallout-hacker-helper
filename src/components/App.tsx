import React from 'react'
import { HashRouter, Route, Routes } from 'react-router-dom'
import PuzzleProvider from './PuzzleProvider/PuzzleProvider'
import WordEntry from './WordEntry/WordEntry'
import WordConfirmation from './WordConfirmation/WordConfirmation'
import GuessProvider from './GuessProvider/GuessProvider'
import Guess from './Guess/Guess'
import Congratulations from './Congratulations/Congratulations'

const App = () => (
  <HashRouter>
    <PuzzleProvider>
      <Routes>
        <Route path="/" element={<WordEntry />} />
        <Route path="/confirm" element={<WordConfirmation />} />
        <Route
          path="/guess"
          element={
            <GuessProvider>
              <Guess />
            </GuessProvider>
          }
        />
        <Route path="/congratulations" element={<Congratulations />} />
      </Routes>
    </PuzzleProvider>
  </HashRouter>
)

export default App
