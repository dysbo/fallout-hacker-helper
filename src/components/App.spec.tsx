import App from './App'
import { render } from '@testing-library/react'

describe('App test', () => {
  test('should render the app', () => {
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })
})
