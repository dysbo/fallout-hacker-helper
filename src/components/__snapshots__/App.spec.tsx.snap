// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test should render the app 1`] = `
<div>
  <div
    class="text"
  >
    Be sure to remove as many duds as possible before you begin.
  </div>
  <div
    class="text"
  >
    Enter the words available in a single line, separated by a space.  When finished, press ENTER to submit them.
  </div>
  <div
    class="user-input"
  >
    <div>
      &gt;Â 
    </div>
    <input
      class="user-input-text"
      type="text"
      value=""
    />
  </div>
</div>
`;
